src/core/thread/mle_router.cpp

```diff
@@ -447,14 +447,47 @@ void MleRouter::HandleAdvertiseTrickleTimer(void)
 
 void MleRouter::StopAdvertiseTrickleTimer(void) { mAdvertiseTrickleTimer.Stop(); }
 
+void MleRouter::RecalculateAdvertiseInterval(void)
+{
+    VerifyOrExit(IsRouterOrLeader());
+
+    {
+        uint32_t neighbors = 0;
+
+        // Count the neighbor routers
+        for (const Router &router : Get<RouterTable>())
+        {
+            if (router.IsStateValid())
+            {
+                neighbors++;
+            }
+        }
+
+        uint32_t advertiseIntervalMax = OT_MIN(kAdvertiseIntervalMax, (neighbors + 1) * 4);
+        advertiseIntervalMax          = OT_MAX(12, advertiseIntervalMax);
```

Comment:
@abtink , should we remove `OT_MIN` and `OT_MAX` from `common/code_utils.hpp`?

Reply:
I think we delayed removing them since they was being used on some platform code (platform should not technically include  `code_utils.hpp` but some do).

Other case they may be useful is if/when defining constants (but I think we updated/replaced all of such cases).